File: Backend/src/EventifyDomain/Event.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace EventifyDomain
{
    public class Event
    {
        public int Id { get; set; }
        public string Theme { get; set; }
        public string Location { get; set; }
        public DateTime? Date { get; set; }
        public int QtyPeople { get; set; }
        public string ImageURL { get; set; }
        public string Phone { get; set; }
        public string Email { get; set; }
        public IEnumerable<Batch> Batches { get; set; }
        public IEnumerable<SocialMedia> SocialMedia { get; set; }

        public IEnumerable<SpeakerEvent> SpeakerEvents { get; set; }
    }
}

File: Frontend/EventifyApp/src/app/models/Event.ts
import { Batch } from "./Batch";
import { SocialMedia } from "./SocialMedia";
import { Speaker } from "./Speaker";

export interface Event {
    id: number;
    theme: string;
    location: string;
    date?: Date;
    qtyPeople: number;
    imageURL: string;
    phone: string;
    email: string;
    batches: Batch[];
    socialMedia: SocialMedia[];
    speakerEvents: Speaker[];
}
